buildSrc/src/main/kotlin/UpdateTranslationsTask.kt

```diff
@@ -0,0 +1,379 @@
+import java.io.File
+import java.io.IOException
+import java.nio.file.Files
+import java.util.*
+import javax.xml.parsers.DocumentBuilderFactory
+import kotlin.concurrent.thread
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.TaskAction
+import org.w3c.dom.Element
+
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A task that checks out an Android repository with string translations, extracts the translations
+ * we're interested in and writes Kotlin source files that provide them.
+ */
+open class UpdateTranslationsTask : DefaultTask() {
+
+    /**
+     * The git binary to use.
+     */
+    @Input
+    var git: String = "git"
+
+    /**
+     * The URL of the repository to check out.
+     */
+    @Input
+    lateinit var gitRepo: String
```

Comment:
It is much better and canonical to use gradle Providers for task properties: https://docs.gradle.org/current/userguide/lazy_configuration.html

Reply:
nm, apparently it's possible to call `property.convention()` from the base class constructor; not sure how that works.